src/suggestion_iterator.cpp

```diff
@@ -51,10 +51,16 @@ SuggestionIterator::SuggestionIterator(const SuggestionIterator& it)
     else if (it.mp_rangeIterator) mp_rangeIterator = std::unique_ptr<RangeIterator>(new RangeIterator(*it.mp_rangeIterator));
 }
 
-SuggestionIterator & SuggestionIterator::operator=(const SuggestionIterator& it) {
-    *mp_rangeIterator = *it.mp_rangeIterator;
-    *mp_internal = *it.mp_internal;
-    *m_suggestionItem = *it.m_suggestionItem;
+SuggestionIterator& SuggestionIterator::operator=(const SuggestionIterator& it) {
+    if (!it.mp_rangeIterator) mp_rangeIterator.reset();
+    else if (!mp_rangeIterator) mp_rangeIterator = std::unique_ptr<RangeIterator>(new RangeIterator(*it.mp_rangeIterator));
+    else *mp_rangeIterator = *it.mp_rangeIterator;
```

Comment:
You can use reset all the time here, whatever is the value of `mp_rangeIterator`.